src/Nox.Types/Types/Color/ColorHelper.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Globalization;
+using System.Linq;
+
+namespace Nox.Types;
+
+public static class ColorHelper
+{
+    public static Color ConvertFromString(string strValue, CultureInfo culture)
+    {
+        string text = strValue.Trim();
+
+        if (text.Length == 0)
+        {
+            return new Color();
+        }
+
+        {
+            System.Drawing.Color color;
+
+            if (KnownColor.TryGetNamedColor(text, out color))
+            {
+                var c = System.Drawing.Color.FromArgb(color.A, color.R, color.G, color.B);
+
+                return Color.From(c.A, c.R, c.G, c.B);
+            }
+        }
+
+        char listSeparator = culture.TextInfo.ListSeparator[0];
+
+        if (!text.Contains(listSeparator))
+        {
+            if ((text.StartsWith("'") && text.EndsWith("'")) ||
+                (text.StartsWith("\"") && text.EndsWith("\"")))
+            {
+                string colorName = text.Trim('\'', '\"');
+                return Color.FromName(colorName);
+            }
+            else if (text.StartsWith("#") && text.Length == 7 ||
+                     (text.StartsWith("0x", StringComparison.OrdinalIgnoreCase) ||
+                      text.StartsWith("&h", StringComparison.OrdinalIgnoreCase)) && text.Length == 8)
+            {
+                try
+                {
+                    int argb = Convert.ToInt32(text.Substring(text[0] == '#' ? 1 : 2), 16);
+                    byte a = (byte)((argb >> 24) & 0xFF);
+                    byte r = (byte)((argb >> 16) & 0xFF);
+                    byte g = (byte)((argb >> 8) & 0xFF);
+                    byte b = (byte)(argb & 0xFF);
+                    return PossibleKnownColor(System.Drawing.Color.FromArgb(a, r, g, b));
+                }
+                catch (Exception)
+                {
+                    return new Nox.Types.Color();
+                }
+            }
+        }
+
+        string[] tokens = text.Split(listSeparator);
+        return tokens.Length switch
+        {
+            1 => PossibleKnownColor(System.Drawing.Color.FromArgb(IntFromString(tokens[0], culture))),
+            3 => PossibleKnownColor(System.Drawing.Color.FromArgb(IntFromString(tokens[0], culture), IntFromString(tokens[1], culture), IntFromString(tokens[2], culture))),
+            4 => PossibleKnownColor(System.Drawing.Color.FromArgb(IntFromString(tokens[0], culture), IntFromString(tokens[1], culture), IntFromString(tokens[2], culture), IntFromString(tokens[3], culture))),
+            _ => new Nox.Types.Color(),
+        };
+    }
+
+    private static Color PossibleKnownColor(System.Drawing.Color color)
```

Comment:
Why do we need the if, the return is the same, ?

`return Color.From(color.A, color.R, color.G, color.B);`